WCF/Shared/Implementation/PlatformContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ServiceModel;
+using System.Text;
+using System.Web;
+
+namespace Microsoft.ApplicationInsights.Wcf.Implementation
+{
+    /// <summary>
+    /// Helper class used to speed up checks for HttpContext
+    /// </summary>
+    internal static class PlatformContext
+    {
+        static readonly bool aspNetCompat = ServiceHostingEnvironment.AspNetCompatibilityEnabled;
+
+        /// <summary>
+        /// Returns the HttpContext for the current thread
+        /// </summary>
+        /// <returns>Null if not running on IIS or not available</returns>
+        internal static HttpContext GetContext()
```

Comment:
**nit peak:** if this code will return the RequestTelemetry, not an HttpContext - you can author the code such a way that System.Web wouldn't even load into the process if not running in compatibility mode. Otherwise - thanks for addressing this